tests/infra/rates.py

```diff
@@ -2,84 +2,110 @@
 # Licensed under the Apache 2.0 License.
 import json
 import infra.proc
+import collections
+from statistics import mean, harmonic_mean, median, pstdev
 
 from loguru import logger as LOG
 
-COMMIT_COUNT_CUTTOF = 20
+COMMIT_COUNT_CUTTOF = 50
 
 
 class TxRates:
     def __init__(self, primary):
         self.get_histogram = False
         self.primary = primary
         self.same_commit_count = 0
-        self.data = {}
+        self.histogram_data = {}
+        self.tx_rates_data = []
+        self.all_metrics = {}
         self.commit = 0
 
-        with open("getTxHist.json", "w") as gtxrf:
-            gtxrf.write('{"id":1,"jsonrpc":"2.0","method":"getTxHist","params":{}}\n')
-        with open("getCommit.json", "w") as gcf:
-            gcf.write('{"id":1,"jsonrpc":"2.0","method":"getCommit","params":{}}\n')
-
     def process_next(self):
-        rv = infra.proc.ccall(
-            "./client",
-            "--host={}".format(self.primary.host),
-            "--port={}".format(self.primary.tls_port),
-            "--ca=networkcert.pem",
-            "userrpc",
-            "--cert=user1_cert.pem",
-            "--pk=user1_privk.pem",
-            "--req=getCommit.json",
-            log_output=False,
-        )
-        print(rv.stdout.decode())
-        result = rv.stdout.decode().split("\n")[1]
-        result = json.loads(result)
-        next_commit = result["result"]["commit"]
-        if self.commit == next_commit:
-            self.same_commit_count += 1
-        else:
-            self.same_commit_count = 0
-
-        self.commit = next_commit
+        with self.primary.user_client(format="json") as client:
+            rv = client.rpc("getCommit", {})
+            result = rv.to_dict()
+            next_commit = result["result"]["commit"]
+            if self.commit == next_commit:
+                self.same_commit_count += 1
+            else:
+                self.same_commit_count = 0
+
+            self.commit = next_commit
 
         if self.same_commit_count > COMMIT_COUNT_CUTTOF:
-            self._get_hist()
+            self._get_metrics()
             return False
         return True
 
     def print_results(self):
-        for key in sorted(self.data.keys()):
-            print(key + " : " + str(self.data[key]))
+        print()
```

Comment:
It would be preferable to implement __str__(), and have places that want to print do something like LOG.info(f"Rates: {tx_rates}")